jetty-util/src/main/java/org/eclipse/jetty/util/PhantomCleaner.java

```diff
@@ -0,0 +1,71 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.io.IOException;
+import java.lang.ref.Cleaner;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class PhantomCleaner
```

Comment:
I know it the JVM class uses "Phantom" as a concept, but I don't like the name here.   `FileGcCleaner` would be better.

Reply:
This is a generic mechanism, not only for files, so `JVMCleaner` or `ReferenceCleaner` or just `Cleaner`.